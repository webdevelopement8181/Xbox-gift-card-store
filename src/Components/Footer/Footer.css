.footer {
  margin-left: -10px;
  background-color: #ABC4AA;
  font-family: Arial, sans-serif;
  margin-bottom: -8px;
  width: 2028px;

}

.footer-container {
  display: flex;
  justify-content: space-around;
  height: 15%;
 
}

.footer-section {
  flex: 1;
  margin-left: 90px;
}

.footer-section h3 {
  font-size: 16px;
  /* margin-bottom: 10px; */
}

.footer-section ul {
  list-style-type: none;
  padding: 0;
}

.footer-section ul li {
  margin-bottom: 8px;
}

.footer-section ul li a {
  text-decoration: none;
  color: #000;
}

.footer-section ul li a:hover {
  text-decoration: underline;
}

.social-icons a {
  margin-right: 10px;
  color: #000;
}

.footer-bottom {
  text-align: center;
  border-top: 1px solid #ddd;
  padding: 10px 0;
  margin-top: 20px;
}

.footer-bottom p {
  margin: 0;
  font-size: 14px;
}

.footer-links {
  list-style-type: none;
  padding: 0;
  margin: 10px 0 0 0;
}

.footer-links li {
  display: inline;
  margin-right: 15px;
}

.footer-links li a {
  text-decoration: none;
  color: #000;
}

.footer-links li a:hover {
  text-decoration: underline;
}
