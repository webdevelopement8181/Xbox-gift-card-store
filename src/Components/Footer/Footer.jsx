import React from "react";
import "./Footer.css";

const Footer = () => {
  return (
    <footer className="footer">
      <div className="footer-container">
        <div className="footer-section">
          <h3>Information</h3>
          <ul>
            <li><a href="#">About Us</a></li>
            <li><a href="#">More Search</a></li>
            <li><a href="#">Blog</a></li>
            <li><a href="#">Testimonials</a></li>
            <li><a href="#">Events</a></li>
          </ul>
        </div>
        <div className="footer-section">
          <h3>Helpful Links</h3>
          <ul>
            <li><a href="#">Services</a></li>
            <li><a href="#">Supports</a></li>
            <li><a href="#">Terms & Condition</a></li>
            <li><a href="#">Privacy Policy</a></li>
          </ul>
        </div>
        <div className="footer-section">
          <h3>Our Services</h3>
          <ul>
            <li><a href="#">Brands list</a></li>
            <li><a href="#">Order</a></li>
            <li><a href="#">Return & Exchange</a></li>
            <li><a href="#">Fashion list</a></li>
            <li><a href="#">Blog</a></li>
          </ul>
        </div>
        <div className="footer-section">
          <h3>Contact Us</h3>
          <ul>
            <li><i className="fa fa-phone"></i> +91 9999 999 999</li>
            <li><i className="fa fa-envelope"></i> youremailid.com</li>
            <li className="social-icons">
              <a href="#"><i className="fa fa-facebook"></i></a>
              <a href="#"><i className="fa fa-google-plus"></i></a>
              <a href="#"><i className="fa fa-twitter"></i></a>
              <a href="#"><i className="fa fa-instagram"></i></a>
            </li>
          </ul>
        </div>
      </div>
      <div className="footer-bottom">
        <p>&copy; 2018 @ company.Ltd. | All Rights Reserved</p>
        <ul className="footer-links">
          <li><a href="#">FAQ</a></li>
          <li><a href="#">Privacy</a></li>
          <li><a href="#">Terms & Condition</a></li>
        </ul>
      </div>
    </footer>
  );
};

export default Footer;
