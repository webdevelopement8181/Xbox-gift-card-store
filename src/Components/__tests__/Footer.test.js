import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from '../Footer/Footer.jsx'; 

describe('Footer component', () => {
  it('renders the footer component', () => {
    render(<Footer />);
    
    // Check for the presence of certain sections
    expect(screen.getByText('Information')).toBeInTheDocument();
    expect(screen.getByText('Helpful Links')).toBeInTheDocument();
    expect(screen.getByText('Our Services')).toBeInTheDocument();
    expect(screen.getByText('Contact Us')).toBeInTheDocument();
  });

  it('renders all links', () => {
    render(<Footer />);
    
    // Check that the links are rendered
    expect(screen.getByText('About Us')).toBeInTheDocument();
    expect(screen.getByText('Services')).toBeInTheDocument();
    expect(screen.getByText('Privacy Policy')).toBeInTheDocument();
    expect(screen.getByText('Brands list')).toBeInTheDocument();
  });

  it('renders contact information', () => {
    render(<Footer />);
    
    // Check that the contact information is rendered
    expect(screen.getByText('+91 9999 999 999')).toBeInTheDocument();
    expect(screen.getByText('youremailid.com')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
