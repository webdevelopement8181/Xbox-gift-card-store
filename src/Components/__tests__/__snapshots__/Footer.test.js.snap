// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="footer-container"
    >
      <div
        class="footer-section"
      >
        <h3>
          Information
        </h3>
        <ul>
          <li>
            <a
              href="#"
            >
              About Us
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              More Search
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Testimonials
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Events
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h3>
          Helpful Links
        </h3>
        <ul>
          <li>
            <a
              href="#"
            >
              Services
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Supports
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Terms & Condition
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Privacy Policy
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h3>
          Our Services
        </h3>
        <ul>
          <li>
            <a
              href="#"
            >
              Brands list
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Order
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Return & Exchange
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Fashion list
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Blog
            </a>
          </li>
        </ul>
      </div>
      <div
        class="footer-section"
      >
        <h3>
          Contact Us
        </h3>
        <ul>
          <li>
            <i
              class="fa fa-phone"
            />
             +91 9999 999 999
          </li>
          <li>
            <i
              class="fa fa-envelope"
            />
             youremailid.com
          </li>
          <li
            class="social-icons"
          >
            <a
              href="#"
            >
              <i
                class="fa fa-facebook"
              />
            </a>
            <a
              href="#"
            >
              <i
                class="fa fa-google-plus"
              />
            </a>
            <a
              href="#"
            >
              <i
                class="fa fa-twitter"
              />
            </a>
            <a
              href="#"
            >
              <i
                class="fa fa-instagram"
              />
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="footer-bottom"
    >
      <p>
        Â© 2018 @ company.Ltd. | All Rights Reserved
      </p>
      <ul
        class="footer-links"
      >
        <li>
          <a
            href="#"
          >
            FAQ
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Privacy
          </a>
        </li>
        <li>
          <a
            href="#"
          >
            Terms & Condition
          </a>
        </li>
      </ul>
    </div>
  </footer>
</DocumentFragment>
`;
